homeassistant/components/braviatv/media_player.py

```diff
@@ -94,30 +93,24 @@ def setup_bravia(config, pin, hass, add_entities):
         request_configuration(config, hass, add_entities)
         return
 
-    try:
-        if ipaddress.ip_address(host).version == 6:
-            mode = "ip6"
-        else:
-            mode = "ip"
-    except ValueError:
-        mode = "hostname"
-    mac = get_mac_address(**{mode: host})
-
     # If we came here and configuring this host, mark as done
     if host in _CONFIGURING:
         request_id = _CONFIGURING.pop(host)
         configurator = hass.components.configurator
         configurator.request_done(request_id)
         _LOGGER.info("Discovery configuration done")
 
+    braviarc = BraviaRC(host)
+    braviarc.connect(pin, CLIENTID_PREFIX, NICKNAME)
+    system_info = braviarc.get_system_info()
```

Comment:
Can this return an empty dict or a dict which does not contain the field for the Mac address resulting in a KeyError in the next line?

Reply:
Good catch. If the device will be offline `connect()` and `get_system_info()` methods rise an exception. I'll fix that.